libc/src/__support/wctype_utils.h

```diff
@@ -29,9 +29,14 @@ LIBC_INLINE cpp::optional<int> wctob(wint_t c) {
   // This needs to be translated to EOF at the callsite. This is to avoid
   // including stdio.h in this file.
   // The standard states that wint_t may either be an alias of wchar_t or
-  // an alias of an integer type, so we need to keep the c < 0 check.
+  // an alias of an integer type where different platforms define this type with
+  // different signedness, so we need to keep the c < 0 check, hence the
+  // pragmas.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtype-limits"
   if (c > 127 || c < 0)
```

Comment:
Would `(c & ~127) != 0` be better? (maybe not straightforward enough)

Reply:
given how simple the condition is, it's cleaner often faster to let the compiler do that sort of optimization. In this specific case, they actually both get compiled to the same result: https://godbolt.org/z/f1E8beTch

Reply:
I think the point is to remove the warnings.  And with that, I think `(c & ~127) != 0` is better than wrapping with `#pragma`'s.  Maybe adding comments to the condition like
```
  This is equivalent to check `(c > 127) || (c < 0)`, but also works without warnings when `wint_t` is unsigned.
```
to make it clearer.